File: Problem10/identical.rb
def is_same_tree(p, q)
    if(p==nil && q==nil)
          return true
    elsif(p==nil or q==nil or p.val!=q.val)
         return false
    end
    check_left=is_same_tree(p.left,q.left)
    check_right=is_same_tree(p.right,q.right)
    return check_left && check_right
end

File: Problem10/identical.java
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if((p==null) && (q==null))
              return true;
        if((p==null) || (q==null) || (p.val!=q.val))
            return false;
        boolean check_left=isSameTree(p.left,q.left);
        boolean check_right=isSameTree(p.right,q.right);
        return (check_left && check_right);
    }
}